File: Frontend_RankingGeKu/src/app/models/gymnastics.ts
/** Geräte in fixer Reihenfolge; Index 0..5 entspricht D1..D6 für Gruppe 1. */
export const APPARATUS = ['Boden', 'Pferd', 'Ring', 'Sprung', 'Barren', 'Reck'];

/** Anzahl Geräte bzw. Durchgänge (D1..D6). */
export const NOTE_COUNT = APPARATUS.length;

/** Kategorie EPA (turnt nicht an Pferd und Ring). */
export function isEpa(kat: string | null | undefined): boolean {
  return (kat ?? '').trim().toUpperCase() === 'EPA';
}

/** Geräte, an denen EPA-Athleten nicht turnen. */
export function isEpaExcludedApparatus(apparatusName: string): boolean {
  return apparatusName === 'Pferd' || apparatusName === 'Ring';
}

File: solBackend_RankingGeKu/Backend_RankingGeKu/Domain/Gymnastics.cs
namespace Backend_RankingGeKu.Domain;

/// <summary>
/// Fachliche Konstanten und Regeln des GeKu-Wettkampfs,
/// die von mehreren Controllern/Services gebraucht werden.
/// </summary>
public static class Gymnastics
{
    /// <summary>Anzahl Geräte bzw. Durchgänge (D1..D6).</summary>
    public const int ApparatusCount = 6;

    /// <summary>Geräte in fixer Reihenfolge (entspricht D1..D6 für Gruppe 1).</summary>
    public static readonly string[] DefaultApparatus =
        { "Boden", "Pferd", "Ring", "Sprung", "Barren", "Reck" };

    /// <summary>Kategorie EPA (turnt nicht an Pferd und Ring).</summary>
    public static bool IsEpa(string? kat) =>
        !string.IsNullOrWhiteSpace(kat) &&
        string.Equals(kat.Trim(), "EPA", StringComparison.OrdinalIgnoreCase);

    /// <summary>Geräte, an denen EPA-Athleten nicht turnen.</summary>
    public static bool IsEpaExcludedApparatus(string? apparatusName) =>
        string.Equals(apparatusName, "Pferd", StringComparison.OrdinalIgnoreCase) ||
        string.Equals(apparatusName, "Ring", StringComparison.OrdinalIgnoreCase);
}
